network.go

```diff
@@ -236,15 +219,56 @@ func (s *sandbox) updateInterface(netHandle *netlink.Handle, iface *pb.Interface
 		// The link is up, makes sure we get it down before
 		// doing any modification.
 		if err := netHandle.LinkSetDown(link); err != nil {
-			return err
+			goto error_case
 		}
 	}
 
-	if err := updateLink(netHandle, link, iface, actionType); err != nil {
-		return err
+	err = updateLink(netHandle, link, iface)
+
+error_case:
+	// in the event that an error occurred during the interface update, make sure we return
+	// the resulting state instead of the requested state
+	if err != nil {
+		resultingIfc = getInterface(netHandle, link)
+	} else {
+		resultingIfc = iface
+	}
+
+	//Put link back into up state
+	retErr := netHandle.LinkSetUp(link)
+
+	// If there was an error updating the interface, give that error precedence
+	// over a potentional LinkSetUp error.
+	if err != nil {
+		retErr = err
+	}
+
+	return resultingIfc, retErr
+}
+
+// getInterface will retrieve interface details from the provided link
+func getInterface(netHandle *netlink.Handle, link netlink.Link) *pb.Interface {
+	var ifc pb.Interface
+	linkAttrs := link.Attrs()
+	ifc.Name = linkAttrs.Name
+	ifc.Mtu = uint64(linkAttrs.MTU)
+	ifc.HwAddr = linkAttrs.HardwareAddr.String()
+
+	addrs, err := netHandle.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		agentLog.WithError(err).Error("getInterface() failed")
+	}
+	for _, addr := range addrs {
+		s := strings.Split(addr.String(), "/")
+
+		m := pb.IpAddress{
+			Address: s[0],
+			Mask:    s[1],
```

Comment:
This looks wrong as `addr.String()` returns a value of the form `ip/mask name` meaning `s[1]` will be `mask name`, not `mask` I think.

Reply:
Yes, updating now...

Reply:
@jodh-intel -- I updated this as well as the test to check the entire struct in the failure test (the test now would've caught this oversight).  Good catch - looks good now.